ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -356,8 +368,51 @@ void fillChunks() {
           if (chunkCounter != 0 && lastChunk.buf.position() == 0) {
             logger.trace("The last buffer(s) received from chunkFillerChannel have no data, discarding them.");
           } else {
-            lastChunk.onFillComplete(true);
-            updateChunkFillerWaitTimeMetrics();
+            if (!passedInBlobProperties.isEncrypted()) {
```

Comment:
This method has become big - could you move the "encryption" part out to a different method to make this more readable? Preferably looks the same as before except for the `if-else` for encrypting.

Reply:
sure. Will do it